pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -224,6 +231,33 @@ func (uc *UpgradeCluster) getClusterNodeStatus(az armhelpers.AKSEngineClient, re
 		}
 	}
 
+	if err := uc.setFaultDomainCount(ctx, resourceGroup, availabilitySetIDs); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (uc *UpgradeCluster) setFaultDomainCount(ctx context.Context, resourceGroup string, vmasIds []string) error {
```

Comment:
This function is copied with only slight modification from cmd/scale.go. I could not find a shared code location that did not cause a circular import, and it seems too specialized to add to `AzureClient` itself. But the duplication still needs to be addressed.

Reply:
Can this be unit tested? same for the one in scale.go

Reply:
I think so—I'm adding tests now—but I'd really like not to duplicate this first, if possible.